polkadot/runtime/parachains/src/assigner_bulk/mod.rs

```diff
@@ -418,18 +429,37 @@ impl<T: Config> Pallet<T> {
 		assignments: Vec<(CoreAssignment, PartsOf57600)>,
 		end_hint: Option<BlockNumberFor<T>>,
 	) -> Result<(), DispatchError> {
-		// There should be at least one assignment
+		// TODO: Add this assert once the calls `request_core_count` and `notify_core_count`
+		// have been established. assert!(core < legacy_core_count + core_count);
```

Comment:
Thank you! We have to subtract legacy cores from the core number before passing it on to the bulk assignment provider. Fixed. With this, you don't have to think about legacy_core_count here at all.